### PR TITLE
Dynamic system for executing validations when the user clicks on the Accept button

### PR DESCRIPTION
**Pull request by joseasoler**

### CODE REVIEW

File: 1.4/Defs/TradeValidationDefs.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Defs>
+	<DynamicTradeInterface.Defs.TradeValidationDef>
+		<defName>DF_InsufficientSilver</defName>
+		<validation>DynamicTradeInterface.UserInterface.TradeValidation.InsufficientSilver:Validate</validation>
```
Comment (Zeracronius) [inline 1.4/Defs/TradeValidationDefs.xml:5]: In the other Defs, i postfixed all the method references with "CallbackHandler"

File: 1.4/Defs/TradeValidationDefs.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+		<text>ConfirmTraderShortFunds</text>
```
Comment (Zeracronius) [inline 1.4/Defs/TradeValidationDefs.xml:6]: what does "text" mean? Translation key?

Comment (Zeracronius) [reply 1.4/Defs/TradeValidationDefs.xml:6]: I haven't used keys directly in xml anywhere else, only for hard-coded values since XML values can be patched.

**Review by Zeracronius — APPROVED** [review]

File: 1.4/Defs/TradeValidationDefs.xml (hunk 2 above)
Comment (joseasoler) [reply 1.4/Defs/TradeValidationDefs.xml:6]: Yes, text is intended to be a translation key. I think it is the best way to go, as the existing confirmation dialogs we are targetting are already using translatable strings for the text.

File: 1.4/Defs/TradeValidationDefs.xml (hunk 1 above)
Comment (joseasoler) [reply 1.4/Defs/TradeValidationDefs.xml]: I have renamed this field to `validationCallbackHandler` in the latest version of the PR.

File: 1.4/Defs/TradeValidationDefs.xml (hunk 2 above)
Comment (joseasoler) [reply 1.4/Defs/TradeValidationDefs.xml]: This field has also been renamed to `textKey` to show explicitly that it is a translation key.

File: 1.4/Schemas/TradeValidationDef.xsd [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+				<xs:element name="DynamicTradeInterface.Defs.TradeValidationDef" type="TradeValidationDef" minOccurs="1" maxOccurs="unbounded" />
+			</xs:sequence>
+		</xs:complexType>
+	</xs:element>
+	<xs:simpleType name="SortDirection">
+		<xs:restriction base="xs:string">
+			<xs:enumeration value="Ascending" />
+			<xs:enumeration value="Descending" />
+		</xs:restriction>
+	</xs:simpleType>
```
Comment (Zeracronius) [inline 1.4/Schemas/TradeValidationDef.xsd:14]: You probably don't need this type definition here

**Comment by Zeracronius** [comment]
For reference, if you don't force push changes then addition modifications you make to the PR as additional commits will be tracked separately and i would be able to see only what has changed since i last looked at the code 😁

**Pull request merged**